e2e-tests-rust/tests/lib.rs

```diff
@@ -872,3 +874,34 @@ async fn test_server_port_fallback() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn storage_at_historical() -> anyhow::Result<()> {
+    // Test that we can query historical storage slots
+    let tester = AnvilZksyncTesterBuilder::default().build().await?;
+
+    tester.tx().finalize().await?;
+    tester.tx().finalize().await?;
+    tester.tx().finalize().await?;
+
+    const NONCE_HOLDER_ADDRESS: Address = address!("0000000000000000000000000000000000008003");
+    let address = B256::left_padding_from(tester.l2_provider().default_signer_address().as_ref());
+    let key = keccak256(address.concat_const::<32, 64>(B256::ZERO));
```

Comment:
There's the helper function to obtain this directly:

```rs
let nonce_key = zksync_types::get_nonce_key(&from_addr)
```

Reply:
I know but I purposefully avoid `zksync_*` dependencies in e2e tests. Just to make sure the majority of flows work even if you stick with alloy and alloy-zksync